Java/syntax_test_java.java

```diff
@@ -746,6 +746,67 @@ int operators() {
   }
 //^ meta.method.java meta.method.body.java punctuation.section.block.end.java
 
+  int numbers() {
+    a = 0 + 0L;
+//      ^ constant.numeric
+//          ^^ constant.numeric
+
+    a = 0xABCD + 0xAB_CD;
+//      ^^^^^^ constant.numeric
+//               ^^^^^^ constant.numeric
+
+    a = 0x_AB_CD - 0xAB_CD_;
+//      ^^^^^^^^ -constant.numeric
+//                 ^^^^^^^^ -constant.numeric
+
+    a = 07 + 0_7;
+//      ^^ constant.numeric
+//           ^^^ constant.numeric
+
+    a = 07_ - 09;
+//      ^^^ -constant.numeric
+//            ^^ -constant.numeric
+
+    a = 0b101101 + 0b10_11_01;
+//      ^^^^^^^^ constant.numeric
+//                 ^^^^^^^^^^ constant.numeric
+
+    a = 0b_101101;
+//      ^^^^^^^^^ -constant.numeric
+
+    a = 12345 + 12_34_5 + 1_____5;
+//      ^^^^^ constant.numeric
+//              ^^^^^^^ constant.numeric
+//                        ^^^^^^^ constant.numeric
+
+    a = 12345l + 12345L + 123_45d + 12_3245F
+//      ^^^^^^ constant.numeric
+//               ^^^^^^ constant.numeric
+//                        ^^^^^^^ constant.numeric
+//                                  ^^^^^^^^ constant.numeric
+
+    a = 12_34_5_ - _12_34_5 - 12_D - 12_L;
```

Comment:
Can you add a test case where there is no space in between the operator.

Reply:
Added.